File: GL_PROJ/Models/DTO/MessageDTO.cs
using System.ComponentModel.DataAnnotations;

namespace GL_PROJ.Models.DTO
{
    public class MessageDTO
    {
        // Custom message Id
        public int MessageId { get; set; }

        // Message text
        public string Data { get; set; }

        // Date the message was sent 
        public DateTime Date { get; set; }

        // Message type
        public int Type { get; set; }

        // Message author Id
        public int UserId { get; set; }

        // Author Name
        public string UserName { get; set; }

        // Group Id
        public int GroupId { get; set; }

    }
}

File: GL_PROJ/ClientApp/src/app/model/MessageDTO.ts
export class MessageDTO {
    data: string;
    date: Date;
    type: number;
    groupId: number;
    messageId: number;
    userId: number;

    constructor(DATA: string, DATE: Date, TYPE: number, GROUP_ID: number, MESSAGE_ID: number, USER_ID: number){
        this.data = DATA;
        this.date = DATE;
        this.type = TYPE;
        this.groupId = GROUP_ID;
        this.messageId = MESSAGE_ID;
        this.userId = USER_ID;
    }
}